src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -569,7 +569,12 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         assert Thread.holdsLock(this);
         ensureOpen();
         settings = getSettings(nodeId, seed, settings);
-        String name = buildNodeName(nodeId);
+        String name;
+        if (settings.get("node.name") != null) {
```

Comment:
can we prevent this. I think we have to lock this naming thing down here and it should be owned by the test cluster